x-pack/plugins/security/server/routes/authentication/saml.ts

```diff
@@ -23,79 +22,47 @@ export function defineSAMLRoutes({
   buildFlavor,
   docLinks,
 }: RouteDefinitionParams) {
-  // Generate two identical routes with new and deprecated URL and issue a warning if route with deprecated URL is ever used.
-  // For a serverless build, do not register deprecated versioned routes
-  for (const path of [
-    '/api/security/saml/callback',
-    ...(buildFlavor !== 'serverless' ? ['/api/security/v1/saml'] : []),
-  ]) {
-    const isDeprecated = path === '/api/security/v1/saml';
-    router.post(
-      {
-        path,
-        security: {
-          authz: {
-            enabled: false,
-            reason: 'This route must remain accessible to 3rd-party SAML providers',
-          },
-        },
-        validate: {
-          body: schema.object(
-            { SAMLResponse: schema.string(), RelayState: schema.maybe(schema.string()) },
-            { unknowns: 'ignore' }
-          ),
-        },
-        options: {
-          access: 'public',
-          excludeFromOAS: true,
-          authRequired: false,
-          xsrfRequired: false,
-          tags: [ROUTE_TAG_CAN_REDIRECT, ROUTE_TAG_AUTH_FLOW],
-          ...(isDeprecated && {
-            deprecated: {
-              documentationUrl: docLinks.links.security.deprecatedV1Endpoints,
-              severity: 'warning',
-              message: i18n.translate('xpack.security.deprecations.samlPostRouteMessage', {
-                defaultMessage:
-                  'The "{path}" URL is deprecated and will be removed in the next major version. Use "/api/security/saml/callback" instead.',
-                values: { path },
-              }),
-              reason: {
-                type: 'migrate',
-                newApiMethod: 'POST',
-                newApiPath: '/api/security/saml/callback',
-              },
-            },
-          }),
+  router.post(
+    {
+      path: '/api/security/saml/callback',
+      security: {
+        authz: {
+          enabled: false,
+          reason: 'This route must remain accessible to 3rd-party SAML providers',
         },
       },
-      async (context, request, response) => {
-        if (isDeprecated) {
-          const serverBasePath = basePath.serverBasePath;
-          logger.warn(
-            // When authenticating using SAML we _expect_ to redirect to the SAML Identity provider.
-            `The "${serverBasePath}${path}" URL is deprecated and might stop working in a future release. Use "${serverBasePath}/api/security/saml/callback" URL instead.`
-          );
-        }
+      validate: {
+        body: schema.object(
+          { SAMLResponse: schema.string(), RelayState: schema.maybe(schema.string()) },
+          { unknowns: 'ignore' }
+        ),
+      },
+      options: {
+        access: 'public',
+        excludeFromOAS: true,
+        authRequired: false,
```

Comment:
__question:__ while you're here, would you mind replacing deprecated `authRequeired` with `security.authc.enabled`?

Reply:
Good call.